src/main/java/com/eternalcode/parcellockers/parcel/gui/ParcelListGUI.java

```diff
@@ -133,12 +137,22 @@ public List<String> replaceParcelPlaceholders(Parcel parcel, List<String> lore)
                 .toString()
             );
 
-        this.lockerRepository.findByUUID(parcel.destinationLocker()).join().ifPresent(locker -> formatter
+        UUID destinationLocker = parcel.destinationLocker();
+        if (!this.lockerRepository.isInCache(destinationLocker)) {
+            this.lockerRepository.findByUUID(destinationLocker).join().ifPresent(locker -> formatter
+                .register("{POSITION_X}", locker.position().x())
+                .register("{POSITION_Y}", locker.position().y())
+                .register("{POSITION_Z}", locker.position().z())
+            );
+        }
+
+        this.lockerRepository.findLocker(destinationLocker).ifPresent(locker -> formatter
             .register("{POSITION_X}", locker.position().x())
             .register("{POSITION_Y}", locker.position().y())
             .register("{POSITION_Z}", locker.position().z())
         );
```

Comment:
Create a method in LockerDatabaseService that will return the optional parcel. It will check whether it is in the cache, if so, it will return it, and if not, it will find it in the database. 
Or use:
```suggestion
        UUID destinationLocker = parcel.destinationLocker();
        this.lockerRepository.findLocker(destinationLocker)
                .or(() -> this.lockerRepository.findByUUID(destinationLocker))
                .ifPresent(locker -> formatter
                       .register("{POSITION_X}", locker.position().x())
                       .register("{POSITION_Y}", locker.position().y())
                       .register("{POSITION_Z}", locker.position().z())
                ))
```
But I think the first suggestion is more readable and simpler.
@Rollczi Should finally decide.

Reply:
There is a bit of playing cat and mouse in this PR. My approach is to make these changes in a separate PR, as I want to merge this as fast as possible and move on forward to the next one to finally deliver this plugin. This PR is on-hold since July, so I will note your change somewhere and make appropriate changes to the database in next PR's. I plan to rewrite the database system to ORMLite in the far future.

Reply:
?
Do you want to release a plugin with bugs as quickly as possible, or do you want to refine the plugin and release it when it works well?

Reply:
You got a good point, yet I still haven't received any opinion from Rollczi, so I guess I have to wait. Is there anything else you want me to improve?

Reply:
Mark Rollczi as a requester.
To answer the question - probably not.

Reply:
No lepiej żeby serwis zarządzał dostępem do danych z cache i repo, a nie gui. Ale zaproponowana zmiana przez DMK jest dziwna. Przepraszam, że piszę po polsku ale jestem już zmęczony, a zostałem zpingowany.

Reply:
Zaproponowana zmiana była opcjonalna. 
Na samym początku napisałem, aby stworzyć w serwisie metodę.